### PR TITLE
feat: Routing trace presenter

### PR DESCRIPTION
**Pull request by joeauyeung**
## What does this PR do?

Adds a routing trace viewer for bookings routed via forms, accessible from the booking actions dropdown. Includes a new tRPC endpoint and presenters that format trace steps into clear, human-readable messages.

- Fixes #XXXX (GitHub issue number)
- Fixes CAL-XXXX (Linear issue number - should be visible at the bottom of the GitHub issue description)

## Visual Demo (For contributors especially)
image
 updated change.

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox. N/A
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

1. Create a booking via a routing form
2. Navigate to the bookings list
3. Click the dropdown menu on a booking that was routed via a form
4. Click "Routing trace" to open the side sheet
5. Verify the trace steps are displayed with domain badges, timestamps, and human-readable messages

Run tests:
```bash
TZ=UTC yarn test packages/features/routing-trace/
TZ=UTC yarn test packages/trpc/server/routers/viewer/bookings/getRoutingTrace.handler.test.ts
```

## Checklist

- [x] I have read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- [x] My code follows the style guidelines of this project
- [x] I have checked if my changes generate no new warnings

## Human Review Checklist

- [ ] Verify the `getRoutingTrace.handler.ts` correctly falls back from permanent RoutingTrace to PendingRoutingTrace
- [ ] Confirm the RoutingTraceSheet UI displays correctly for bookings with routing traces
- [ ] Check that the handler returns empty steps gracefully when no trace exists
- [ ] Verify the authorization check via `BookingAccessService` properly restricts access

---

## Updates since last revision

**Latest fix (test updates):**
- Updated `getRoutingTrace.handler.test.ts` to include `ctx` parameter with mock user
- Added mock for `BookingAccessService` to match the handler's authorization check
- Fixed `@calcom/prisma` mock to include both `default` and `prisma` exports (resolves vitest mock error)

**Previous updates:**
- Added comprehensive test coverage:
  - **`PrismaRoutingTraceRepository.test.ts`**: Added 3 tests for the new `findByBookingUid` method
  - **`getRoutingTrace.handler.test.ts`**: 6 tests covering permanent routing trace, fallback to PendingRoutingTrace, missing data handling, and edge cases

---

## Summary by cubic
Adds a routing trace viewer for bookings routed via forms, accessible from the booking actions dropdown. Includes a new tRPC endpoint and presenters that format trace steps into clear, human-readable messages.

- **New Features**
  - "Routing trace" action in the booking dropdown opens a side sheet timeline with domain badges, icons, skeleton loading, and millisecond timestamps.
  - New endpoint: viewer.bookings.getRoutingTrace retrieves a trace by booking UID and falls back to a pending trace from the form response.
  - Domain-specific presenters for Salesforce and Routing Form, with a core presenter that delegates and returns readable step messages.
  - i18n key added: routing_trace.

- **Refactors**
  - Repository adds findByBookingUid; booking actions store adds isOpenRoutingTraceSheet.
  - Unit tests cover all presenters, repository methods, and the tRPC handler.

---

**Link to Devin run**: https://app.devin.ai/sessions/90dcf57bbce34119a6367b402f0dc1df
**Requested by**: joe@cal.com (@joeauyeung)

### CODE REVIEW

File: apps/web/components/booking/RoutingTraceSheet.tsx
Comment (joeauyeung) [inline apps/web/components/booking/RoutingTraceSheet.tsx:1]: The way we present the routing trace isn't set in stone. We can easily change this based on feedback

File: packages/features/routing-trace/presenters/RoutingFormTracePresenter.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import type { RoutingStep } from "../repositories/RoutingTraceRepository.interface";
+
+export class RoutingFormTracePresenter {
```
Comment (joeauyeung) [inline packages/features/routing-trace/presenters/RoutingFormTracePresenter.ts:3]: Each domain will have it's own presenter class that handles how the data stored is displayed to the user. 

By having a presenter class it gives us flexibility on changing how the data is presented.

File: packages/features/routing-trace/presenters/RoutingTracePresenter.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import { SalesforceRoutingTracePresenter } from "./SalesforceRoutingTracePresenter";
+
+export interface PresentedStep {
+  message: string;
+  domain: string;
+  step: string;
+  timestamp: number;
+}
+
+export class RoutingTracePresenter {
```
Comment (joeauyeung) [inline packages/features/routing-trace/presenters/RoutingTracePresenter.ts:13]: Core `RoutingTracePresenter` calls on the domain specific presenters

File: packages/trpc/server/routers/viewer/bookings/getRoutingTrace.handler.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    include: { pendingRoutingTrace: true },
+  });
+
+  if (formResponse?.pendingRoutingTrace?.trace) {
+    const trace = formResponse.pendingRoutingTrace.trace as unknown as RoutingTrace;
+    return { steps: RoutingTracePresenter.present(trace) };
+  }
+
+  return { steps: [] };
+};
```
[the remark this answers is not in the document]
Comment (joeauyeung) [reply packages/trpc/server/routers/viewer/bookings/getRoutingTrace.handler.ts]: In the _router this is an authed procedure. Does that resolve your comment?

File: packages/features/routing-trace/presenters/getDomainLabel.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { ROUTING_TRACE_DOMAINS } from "../constants";
+
+const DOMAIN_LABELS: Record<string, string> = {
+  [ROUTING_TRACE_DOMAINS.SALESFORCE]: "Salesforce",
```
[the remark this answers is not in the document]
Comment (joeauyeung) [reply packages/features/routing-trace/presenters/getDomainLabel.ts:4]: This is interesting because there's so much variables that go into the string. Let me follow up with this @volnei

**Review by volnei — DISMISSED** [review]

**Review by volnei — APPROVED** [review]
